src/openms/include/OpenMS/FORMAT/HANDLERS/MzMLHandler.h

```diff
@@ -693,24 +693,39 @@ namespace OpenMS
           //if defaultArrayLength > 0 : warn that no m/z or int arrays is present
           if (default_arr_length != 0)
           {
-            warning(LOAD, String("The m/z or intensity array of chromatogram '") + inp_chromatogram.getNativeID() + "' is missing and default_arr_length is " + default_arr_length + ".");
+            warning(LOAD, String("The m/z or intensity array of chromatogram '") +
+                inp_chromatogram.getNativeID() + "' is missing and default_arr_length is " + default_arr_length + ".");
           }
           return;
         }
 
-        //Warn if the decoded data has a different size than the defaultArrayLength
+        // Warn if the decoded data has a different size than the defaultArrayLength
         Size rt_size = rt_precision_64 ? input_data[rt_index].floats_64.size() : input_data[rt_index].floats_32.size();
+        Size int_size = int_precision_64 ? input_data[int_index].floats_64.size() : input_data[int_index].floats_32.size();
+        // Check if int-size and rt-size are equal
+        if (rt_size != int_size)
+        {
+          fatalError(LOAD, String("The length of RT and intensity values of chromatogram '") + inp_chromatogram.getNativeID() + "' differ (rt-size: " + rt_size + ", int-size: " + int_size + "! Not reading chromatogram!");
+        }
+        bool repair_array_length = false;
         if (default_arr_length != rt_size)
         {
           warning(LOAD, String("The base64-decoded rt array of chromatogram '") + inp_chromatogram.getNativeID() + "' has the size " + rt_size + ", but it should have size " + default_arr_length + " (defaultArrayLength).");
+          repair_array_length = true;
         }
-        Size int_size = int_precision_64 ? input_data[int_index].floats_64.size() : input_data[int_index].floats_32.size();
         if (default_arr_length != int_size)
         {
           warning(LOAD, String("The base64-decoded intensity array of chromatogram '") + inp_chromatogram.getNativeID() + "' has the size " + int_size + ", but it should have size " + default_arr_length + " (defaultArrayLength).");
+          repair_array_length = true;
+        }
+        // repair size of array, accessing memory that is beyond int_size will lead to segfaults later
+        if (repair_array_length)
+        {
+          default_arr_length = std::min(int_size, rt_size);
```

Comment:
this does not make sense.
We asserted already that int_size == rt_size.
Maybe
`default_arr_length  = int_size; // set to length of actual data (int_size and rt_size are equal, s.a.)`